src/components/MdDatepicker/MdDatepicker.vue

```diff
@@ -93,6 +94,12 @@
       }
     },
     methods: {
+      onInput: debounce(function (value) {
+        const parsedDate = parse(value)
+        if (isValid(parsedDate)) {
+          this.selectedDate = parsedDate
+        }
+      }, 1000),
```

Comment:
Debounce should be used in this way to make scope in the instance instead of in the component.

```javascript
export default {
  // ...

  methods: {
    onInput (value) {
      const parsedDate = parse(value)
      if (isValid(parsedDate)) {
        this.selectedDate = parsedDate
      }
    }
  }

  created () {
    this.onInput = debounce(this.onInput, 1000)
  }

  // ...
}
```

You could try this example to know the difference.

```vue
<template>
  <div>
    <debounce-test></debounce-test>
    <debounce-test></debounce-test>
  </div>
</template>

<script>
  import MdDebounce from 'core/utils/MdDebounce'

  export default {
    name: 'Debounce scope compare',

    components: {
      debounceTest: {
        name: 'DebounceTest',

        data () {
          return {
            initTime: null,
            wrongDebounceResult: [],
            rightDebounceResult: []
          }
        },

        methods: {
          wrongDebounce: MdDebounce(function () {
            this.wrongDebounceResult.push(new Date() - this.initTime)
          }, 1000),

          rightDebounce () {
            this.rightDebounceResult.push(new Date() - this.initTime)
          },

          exec () {
            this.wrongDebounce()
            this.rightDebounce()
          }
        },

        render (createElement) {
          return createElement('div', 
            [
              createElement(
                'div', `wrong: ${this.wrongDebounceResult}`
              ),
              createElement(
                'div', `right: ${this.rightDebounceResult}`
              )
            ]
          )
        },

        created () {
          this.rightDebounce = MdDebounce(this.rightDebounce, 1000)
          this.initTime = new Date()

          setTimeout(this.exec(), 1000)
        },
      }
    }
  }
</script>
```

Result:
```
wrong:
right: 1345
wrong: 1360
right: 1371
```

Reply:
Hey!

I'm all for your implementation - I also see that this allows the debounce time to be set as a prop rather than hardcoded at 1000ms.

Would you mind explaining why the results are wrong though? I don't quite understand your example.

Reply:
As I said scope shouldn't in the compnent, it should be in the instance. The wrong one has side effect that the second instance use the same `timeout` with the first one. It is expected to execute the method after the last call with debounce per instance. There shouldn't be side effect cross the instances.
For example, if I use a longer delay, or my device performance is bad, or I focus and type in another datepicker quickly, the previous datepicker `onInput` will not be called.